pkg/api/addons.go

```diff
@@ -11,9 +11,8 @@ import (
 	"github.com/Azure/aks-engine/pkg/api/common"
 )
 
-func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
+func (cs *ContainerService) setAddonsConfig(isUpdate bool, cloudSpecConfig AzureEnvironmentSpecConfig) {
 	o := cs.Properties.OrchestratorProfile
-	cloudSpecConfig := cs.GetCloudSpecConfig()
```

Comment:
why this change? I would argue it's cleaner to get the CloudSpecConfig from cs than passing it in as a separate arg.

Reply:
hmm reading into this more I think I understand why you did this. If we don't want to "init" the cloudSpecConfig multiple times, would a variable work better? `GetCloudSpecConfig()` would check if `cloudSpecConfig` is already defined, if not call `SetCloudSpecConfig` (which would be the logic your current `GetCloudSpecConfig` func has, and then return it. ie:

```
func (cs *ContainerService) GetCloudSpecConfig() AzureEnvironmentSpecConfig {
    if cloudSpecConfig == nil {
        setCloudSpecConfig()
   }
    return cloudSpecConfig

} 
```

Reply:
Also, the current code is already calling `GetCloudSpecConfig()` in multiple times in other places so if the idea is to only set the config once you'll need to change all these calls as well.